### PR TITLE
chore: Move e2e into an app

### PR DESCRIPTION
**Pull request by rahulnr7**
Move the framework and all related files of the e2e tests using the detox+jest setup to a package

### CODE REVIEW

File: apps/wallet-mobile/ios/yoroi.xcodeproj/xcshareddata/xcschemes/nightly.xcscheme [hunk 1]
```diff
@@ -30,7 +30,7 @@
       shouldAutocreateTestPlan = "YES">
    </TestAction>
    <LaunchAction
-      buildConfiguration = "Release"
+      buildConfiguration = "Debug"
```
Comment (michalszorad) [inline apps/wallet-mobile/ios/yoroi.xcodeproj/xcshareddata/xcschemes/nightly.xcscheme]: Why is there a change?

File: apps/wallet-mobile/ios/yoroi.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -999,6 +999,7 @@
 				COPY_PHASE_STRIP = YES;
 				ENABLE_NS_ASSERTIONS = NO;
 				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				EXCLUDED_ARCHS = "";
```
Comment (michalszorad) [inline apps/wallet-mobile/ios/yoroi.xcodeproj/project.pbxproj]: Why is there a change?

File: packages/e2e/package.json [hunk 3]
```diff
@@ -0,0 +1,90 @@
+{
+  "name": "@yoroi/e2e",
```
Comment (michalszorad) [inline packages/e2e/package.json]: Can we please also add a README.md explaining how to run the tests locally?

File: apps/wallet-mobile/ios/yoroi.xcodeproj/xcshareddata/xcschemes/nightly.xcscheme (hunk 1 above)
Comment (rahulnr7) [reply apps/wallet-mobile/ios/yoroi.xcodeproj/xcshareddata/xcschemes/nightly.xcscheme]: Thanks for pointing this out. This is not needed.

File: apps/wallet-mobile/ios/yoroi.xcodeproj/project.pbxproj (hunk 2 above)
Comment (rahulnr7) [reply apps/wallet-mobile/ios/yoroi.xcodeproj/project.pbxproj]: Thanks for pointing out. This is not needed as well.

File: packages/e2e/package.json (hunk 3 above)
Comment (rahulnr7) [reply packages/e2e/package.json]: Sure will add. Thanks

File: packages/e2e/package.json [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
```
Comment (stackchain) [inline packages/e2e/package.json]: please create a folder `/tests/` -> `e2e-wallet-mobile`
the packages folder is for dependencies only
```suggestion
  "name": "@yoroi/e2e-wallet-mobile",
```

File: packages/e2e/stepsToRunE2eTests.md
Comment (stackchain) [inline packages/e2e/stepsToRunE2eTests.md:1]: maybe rename it README.md ?

File: packages/e2e/package.json (hunk 4 above)
Comment (rahulnr7) [reply packages/e2e/package.json]: Done

File: packages/e2e/stepsToRunE2eTests.md
Comment (rahulnr7) [reply packages/e2e/stepsToRunE2eTests.md:1]: Done

**Review by stackchain — APPROVED** [review]

**Pull request merged**